src/expm_frechet.jl

```diff
@@ -0,0 +1,532 @@
+
+@doc raw"""
+based on
+.. [1] Awad H. Al-Mohy and Nicholas J. Higham (2009)
+           Computing the Frechet Derivative of the Matrix Exponential,
+           with an application to Condition Number Estimation.
+           SIAM Journal On Matrix Analysis and Applications.,
+           30 (4). pp. 1639-1657. ISSN 1095-7162
+"""
+
+ell_table_61 = (
```

Comment:
what are these magic numbers? They do not even have a comment what they are for?

Reply:
Added comments - there are some typos - will change.

# From table 6.1 of [^AlMohyHigham2009]. A list of cut off values $l_m$ used to determine how many terms of Padé approximant is used to compute expm_frechet(A, E), look for the largest i\
   ndex $m\in [3, 5, 7, 9, 13]$ just exceed $|A|_1$.
ell_table_61 = (

Reply:
The comment in the code contains a few typos, but I am really not happy with such magic numbers somewhere in the code. And especially a Nothing upfront and such.

The problem is, that we as maintainers have to be able to take care of this code. And even in the functions below there is quite a few “magic numbers”. I am really not happy about this.

Reply:
Instead of the table I can write a function implementing the cutoff logic of how many pade terms to use. Will have to think about this. It may not be straightforward, otherwise it is already done in that paper.

Pade coefficients can be generated from a function.  I can construct a expm_pade object, and we initialize it with coefficients then pass to functions that call expm_frechet - that way we dont have to save the coefficients, and dont have to keep regenerating them - it may be a bit more complicated to use the function, though. Appreciate if you have a suggestion.